### PR TITLE
AccountCircleIllustration -> TypeScript

### PR DESCRIPTION
**Pull request by JamesIves**
Updates the `AccountCircleIllustration` to use Typescript. Running into some issues with this, mostly to do with aliased imports in tests though.

I've tried the following configs (this component uses `~constants/js/colors.js`)

* The following config errors the Typescript compiler in tests.
```
    "baseUrl": "./",                  
    "paths": {
      "~components": ["components"],
      "~proptypes": ["proptypes"],
      "~constants": ["constants"]
    },          
```

* The following errors Babel by the looks of things (It also silences intellisense)
```
    "baseUrl": "./",                  
    "paths": {
      "~components": ["components"],
      "~proptypes": ["proptypes"],
      "~constants/js/*": ["constants/js/*"]
    },          
```

The module aliasing occurs in the `.babelrc` configuration previously. 

* Run `yarn start` and give it a try.

I'm going to keep trying to fix this later, but if anyone else wants to take a stab please feel free. I have a few other things I need to wrap up today before I can dedicate more time to this.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-*)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -68,5 +72,5 @@
     "./webpack.config.js",
     "./_templates/",
     "./dist/"
-  ]
+  ],
```
Comment (CalMlynarczyk) [inline tsconfig.json]: Trailing commas aren't valid in JSON.

File: components/atoms/illustrations/AccountCircleIllustration/AccountCircleIllustration.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  height?: string;
+  /** The color of the illustration. */
+  fill?: string;
+  /** Additional class names to apply to the container. */
+  className?: string;
+  /** Additional inline styles to apply to the container. */
+  style?: React.CSSProperties;
+};
+
+const AccountCircleIllustration = ({fill, width, height, className, style}: AccountCircleIllustrationProps) => (
```
[the remark this answers is not in the document]
Comment (JamesIves) [reply components/atoms/illustrations/AccountCircleIllustration/AccountCircleIllustration.tsx]: I'll make this change in the docgen samples too

File: tsconfig.json (hunk 1 above)
Comment (JamesIves) [reply tsconfig.json]: Oops didn't mean to commit this

**Comment by JamesIves** [comment]
@CalMlynarczyk Registering this package in the `testSingle` command fixes it for single tests, but throws an error saying it can't find `Box.tsx` when running all tests as a whole. 

```
/Users/james/Documents/Dev/components/node_modules/typescript/lib/typescript.js:131064
                var error = new Error("Could not find source file: '" + fileName + "'.");
                            ^
Error: Could not find source file: '/Users/james/Documents/Dev/components/components/atoms/Box/Box.tsx'.
```

I assume this is only because `Box` is the first component to be tested. I've pushed up my attempts.

**Comment by JamesIves** [comment]
Closing this in favor of #1070

**Pull request closed without merging**